core/server/services/mega/mega.js

```diff
@@ -228,14 +243,24 @@ async function listener(emailModel, options) {
     }
 }
 
+const statusChangeListener = (emailModel, options) => {
+    const emailRetried = emailModel.wasChanged() && (emailModel.get('status') === 'pending') && (emailModel.previous('status') === 'failed');
+
+    if (emailRetried) {
+        listener(emailModel, options);
```

Comment:
Keeping this retry logic in a listener to behave in similar async fashion as the main listener does.